src/pydicom/dataset.py

```diff
@@ -15,38 +15,37 @@
             contains its own DataElements, and so on in a recursive manner.
 """
 import copy
-from bisect import bisect_left
-from contextlib import nullcontext
 import io
-from importlib.util import find_spec as have_package
-from itertools import takewhile
 import json
 import os
 import os.path
 import re
+import warnings
+import weakref
+from bisect import bisect_left
+from collections.abc import (
+    ValuesView,
+    Iterator,
+    Callable,
+    MutableSequence,
+    MutableMapping,
+    Set,
+)
+from contextlib import nullcontext
+from importlib.util import find_spec as have_package
+from itertools import takewhile
 from types import TracebackType
 from typing import (
     Optional,
     TypeAlias,
-    Union,
     Any,
+    AnyStr,
     cast,
     BinaryIO,
-    AnyStr,
     TypeVar,
     overload,
     TYPE_CHECKING,
 )
-from collections.abc import (
-    ValuesView,
-    Iterator,
-    Callable,
-    MutableSequence,
-    MutableMapping,
-    Set,
-)
-import warnings
-import weakref
```

Comment:
There have been a couple of unused imports, so I did an automatic optimize, which did some reordering...